packages/interface/src/contexts/Round.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { createContext, useContext, useMemo, useCallback, useEffect, useState } from "react";
+
+import { config } from "~/config";
+import { api } from "~/utils/api";
+
+import type { RoundContextType, RoundProviderProps, IRoundData } from "./types";
+import type { IRoundMetadata } from "~/utils/types";
+
+export const RoundContext = createContext<RoundContextType | undefined>(undefined);
+
+export const RoundProvider: React.FC<RoundProviderProps> = ({ children }: RoundProviderProps) => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [rounds, setRounds] = useState<IRoundData[]>([]);
+
+  const polls = api.maci.poll.useQuery(undefined, { enabled: Boolean(config.maciSubgraphUrl) });
+
+  // on load we fetch the data from the poll
+  useEffect(() => {
+    if (polls.data) {
+      return;
+    }
+
+    // eslint-disable-next-line no-console
+    polls.refetch().catch(console.error);
+  }, [polls]);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    if (config.maciSubgraphUrl) {
+      const pollsList = polls.data ? polls.data : undefined;
+
+      if (!pollsList || pollsList.length === 0) {
+        setIsLoading(false);
+        return;
+      }
+
+      const roundsTemp: IRoundData[] = [];
+
+      pollsList.forEach((poll) => {
+        fetch(poll.metadataUrl)
+          .then((res) => res.json())
+          .then((res) => {
+            const r = res as IRoundMetadata;
+            const round = {
+              isMerged: poll.isMerged,
+              pollId: poll.id,
+              duration: poll.duration,
+              deployTime: poll.deployTime,
+              numSignups: poll.numSignups,
+              pollAddress: poll.address,
+              mode: poll.mode,
+              registryAddress: poll.registryAddress,
+              roundId: r.roundId,
+              description: r.description,
+              startsAt: r.startsAt,
+              registrationEndsAt: r.registrationEndsAt,
+              votingStartsAt: r.votingStartsAt,
+              votingEndsAt: r.votingEndsAt,
+              tallyFile: r.tallyFile,
+            } as IRoundData;
+
+            roundsTemp.push(round);
+          })
+          // eslint-disable-next-line no-console
+          .catch(() => console.error);
```

Comment:
This logic should be moved to a separate layer (use redux or effector).
This creates additional complexity and it will be hard to fix this in the future.

Reply:
I move this logic to the trpc server to keep consistency with other methods.